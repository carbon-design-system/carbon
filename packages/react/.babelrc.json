{
  "exclude": ["node_modules/**"],
  "presets": [
    [
      "@babel/preset-env",
      {
        "modules": false,
        "targets": {
          "browsers": ["extends browserslist-config-carbon"]
        }
      }
    ],
    "@babel/preset-react",
    "@babel/preset-typescript"
  ],
  "plugins": [
    "dev-expression",
    "@babel/plugin-proposal-export-default-from",
    "@babel/plugin-transform-class-properties",
    "@babel/plugin-transform-export-namespace-from",
    "@babel/plugin-transform-react-constant-elements"
  ]
}
