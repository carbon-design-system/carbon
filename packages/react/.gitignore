# Storybook
storybook-static

# Generated metadata
react-docgen.json

# Cypress artifacts
cypress/videos
cypress/screenshots