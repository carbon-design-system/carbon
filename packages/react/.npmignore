**/__mocks__/**
**/__tests__/**
**/examples/**
**/tasks/**
**/*-test.js
