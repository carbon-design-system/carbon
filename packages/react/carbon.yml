# yaml-language-server: $schema=https://unpkg.com/@carbon-platform/schemas@v1/carbon-resources.schema.json
library:
  id: carbon-react
  name: Carbon React
  description: React implementation of Carbon Components.
  externalDocsUrl: https://react.carbondesignsystem.com
  inherits: carbon-styles
assets:
    accordion:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/accordion/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-accordion
    aspect-ratio:
      status: stable
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-aspectratio
    breadcrumb:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/breadcrumb/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-breadcrumb
    button:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/button/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-button
    checkbox:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/checkbox/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-checkbox
    code-snippet:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/code-snippet/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-codesnippet
    combo-box:
      status: stable
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-combobox
    composed-modal:
      name: Composed modal
      framework: react
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/composed-modal.svg
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-composedmodal
      tags:
        - input-control
    content-switcher:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/content-switcher/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-contentswitcher
    copy-button:
      status: stable
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-copybutton
    data-table:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/data-table/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-datatable
    date-picker:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/date-picker/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-datepicker
    dropdown:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/dropdown/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-dropdown
    error-boundary:
      name: Error boundary
      framework: react
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/error-boundary.svg 
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-errorboundary
      tags:
        - system-feedback
    file-uploader:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/file-uploader/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-fileuploader--file-uploader
    flex-grid:
      name: Flex grid
      framework: react
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/flex-grid.svg 
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-flexgrid
      tags:
        - shell
    form:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/form/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-form
    form-group:
        name: Form group
        framework: react
        status: stable
        type: component
        platform: web
        thumbnailPath: ./thumbnails/form-group.svg
        demoLinks:
          - type: storybook
            name: Storybook
            action: link
            url: https://react.carbondesignsystem.com/?path=/docs/components-formgroup
        tags:
          - form
    icon-button:
        name: Icon button
        framework: react
        status: stable
        type: component
        platform: web
        thumbnailPath: ./thumbnails/icon-button.svg 
        demoLinks:
          - type: storybook
            name: Storybook
            action: link
            url: https://react.carbondesignsystem.com/?path=/docs/components-iconbutton
        tags:
          - input-control
          - media
    inline-loading:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/inline-loading/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-inlineloading
    layer:
      name: Layer
      framework: react
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/layer.svg 
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-layer
      tags:
        - shell 
    link:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/link/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-link
    loading:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/loading/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-loading
    modal:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/modal/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-modal
    multiselect:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/dropdown/usage/#multiselect
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-multiselect
    notification:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/notification/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-notifications
    number-input:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/number-input/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-numberinput
    ordered-list:
      name: Ordered list
      framework: react
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/ordered-list.svg
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-orderedlist
      tags:
        - data-display
    overflow-menu:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/overflow-menu/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-overflowmenu
    pagination:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/pagination/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-pagination
    pagination-nav:
      status: stable
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-paginationnav
    popover:
      status: stable
      framework: react
      externalDocsUrl: https://www.carbondesignsystem.com/components/popover/usage
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-popover
    progress-bar:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/progress-bar/usage 
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-progressbar
    progress-indicator:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/progress-indicator/usage 
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-progressindicator
    radio-button: 
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/radio-button/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-radiobutton
    search:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/search/usage 
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-search
    select:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/select/usage 
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-select
    skeleton:
      name: Skeleton
      framework: react
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/skeleton.svg
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-skeleton
      tags:
        - shell
    slider:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/slider/usage 
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-slider
    structured-list:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/structured-list/usage 
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-structuredlist
    tabs:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/tabs/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-tabs
    tag:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/tag/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-tag
    text-area:
      status: stable
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-textarea
    text-input:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/text-input/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-textinput
    theme:
      status: stable
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-theme--default
    tile:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/tile/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-tile
    time-picker:
      status: stable
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-timepicker
    toggle:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/toggle/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-toggle
    toggletip:
      name: Toggletip
      framework: react
      externalDocsUrl: https://www.carbondesignsystem.com/components/toggletip/usage
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/toggletip.svg
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-toggletip
      tags:
        - content-element 
    tooltip:
      status: stable
      externalDocsUrl: https://www.carbondesignsystem.com/components/tooltip/usage
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-tooltip
    tooltip-definition:
      status: stable
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-definitiontooltip
    ui-shell-header:
      framework: react
      externalDocsUrl: https://www.carbondesignsystem.com/components/UI-shell-header/usage
      status: stable
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-ui-shell
    unordered-list:
      name: Unordered list
      framework: react
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/unordered-list.svg 
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/components-unorderedlist
      tags:
        - data-display 
    grid:
      status: stable
      type: component
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/elements-grid
    fluid-form:
      name: Fluid form
      framework: react
      status: experimental
      type: component
      platform: web
      thumbnailPath: ./thumbnails/fluid-form.svg
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/experimental-fluidform
      tags:
        - form
    unstable-heading:
      name: Unstable heading
      framework: react
      status: experimental
      type: component
      platform: web
      thumbnailPath: ./thumbnails/unstable-heading.svg  
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-heading
      tags:
        - content-element
    unstable-menu:
      name: Unstable menu
      framework: react
      status: experimental
      type: component
      platform: web
      thumbnailPath: ./thumbnails/unstable-menu.svg
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-menu
      tags:
        - system-feedback
    unstable-pagination:
      name: Unstable pagination
      framework: react
      status: experimental
      type: component
      platform: web
      thumbnailPath: ./thumbnails/unstable-pagination.svg 
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-pagination
      tags:
        - data-display
        - contextual-navigation
    unstable-text:
      name: Unstable text
      framework: react
      status: experimental
      type: component
      platform: web
      thumbnailPath: ./thumbnails/unstable-text.svg
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-text
      tags:
        - input-control
    tree-view:
      status: experimental
      framework: react
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-treeview
    stack:
      name: Stack
      framework: react
      status: stable
      type: component
      platform: web
      thumbnailPath: ./thumbnails/stack.svg  
      demoLinks:
        - type: storybook
          name: Storybook
          action: link
          url: https://react.carbondesignsystem.com/?path=/docs/layout-stack
      tags:
        - content-block
    use-feature-flag:
      name: unstable_useFeatureFlag
      framework: react
      status: experimental
      type: function
      platform: web
      tags:
        - hook
    use-feature-flags:
      name: unstable_useFeatureFlags
      framework: react
      status: experimental
      type: function
      platform: web
      tags:
        - hook
    use-context-menu:
      name: unstable_useContextMenu
      framework: react
      status: experimental
      type: function
      platform: web
      tags:
        - hook
    use-theme:
      name: useTheme
      framework: react
      status: stable
      type: function
      platform: web
      tags:
        - hook
    
