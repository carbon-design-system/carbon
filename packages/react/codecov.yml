coverage:
  status:
    project:
      components:
        target: 100%
        threshold: 1%
        paths:
          - src/components/
