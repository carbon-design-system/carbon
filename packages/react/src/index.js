/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export Accordion from './components/Accordion';
export AccordionItem from './components/AccordionItem';
export { AspectRatio } from './components/AspectRatio';
export { Breadcrumb, BreadcrumbItem } from './components/Breadcrumb';
export Button from './components/Button';
export ButtonSet from './components/ButtonSet';
export Checkbox from './components/Checkbox';
export CodeSnippet from './components/CodeSnippet';
export ComboBox from './components/ComboBox';
export ComposedModal, {
  ModalHeader,
  ModalBody,
  ModalFooter,
} from './components/ComposedModal';
export ContentSwitcher from './components/ContentSwitcher';
export Copy from './components/Copy';
export CopyButton from './components/CopyButton';
export DangerButton from './components/DangerButton';
export {
  default as DataTable,
  Table,
  TableActionList,
  TableBatchAction,
  TableBatchActions,
  TableBody,
  TableCell,
  TableContainer,
  TableExpandHeader,
  TableExpandRow,
  TableExpandedRow,
  TableHead,
  TableHeader,
  TableRow,
  TableSelectAll,
  TableSelectRow,
  TableToolbar,
  TableToolbarAction,
  TableToolbarContent,
  TableToolbarSearch,
  TableToolbarMenu,
} from './components/DataTable';
export DatePicker from './components/DatePicker';
export DatePickerInput from './components/DatePickerInput';
export Dropdown from './components/Dropdown';
export {
  ErrorBoundary,
  ErrorBoundaryContext,
} from './components/ErrorBoundary';
export FileUploader, {
  Filename,
  FileUploaderButton,
  FileUploaderDropContainer,
  FileUploaderItem,
} from './components/FileUploader';
export Form from './components/Form';
export FluidForm from './components/FluidForm';
export FormGroup from './components/FormGroup';
export FormItem from './components/FormItem';
export FormLabel from './components/FormLabel';
export { Grid, Row, Column } from './components/Grid';
export Icon from './components/Icon';
export InlineLoading from './components/InlineLoading';
export Link from './components/Link';
export ListItem from './components/ListItem';
export Loading from './components/Loading';
export Modal from './components/Modal';
export ModalWrapper from './components/ModalWrapper';
export MultiSelect from './components/MultiSelect';
export {
  ToastNotification,
  InlineNotification,
  NotificationActionButton,
  NotificationButton,
  NotificationTextDetails,
} from './components/Notification';
export NumberInput from './components/NumberInput';
export OrderedList from './components/OrderedList';
export OverflowMenu from './components/OverflowMenu';
export OverflowMenuItem from './components/OverflowMenuItem';
export Pagination from './components/Pagination';
export PaginationNav from './components/PaginationNav';
export PrimaryButton from './components/PrimaryButton';
export {
  ProgressIndicator,
  ProgressStep,
} from './components/ProgressIndicator';
export RadioButton from './components/RadioButton';
export RadioButtonGroup from './components/RadioButtonGroup';
export Search from './components/Search';
export SearchFilterButton from './components/SearchFilterButton';
export SearchLayoutButton from './components/SearchLayoutButton';
export ExpandableSearch from './components/ExpandableSearch';
export SecondaryButton from './components/SecondaryButton';
export Select from './components/Select';
export SelectItem from './components/SelectItem';
export SelectItemGroup from './components/SelectItemGroup';
export Switch from './components/Switch';
export Slider from './components/Slider';
export {
  StructuredListWrapper,
  StructuredListHead,
  StructuredListBody,
  StructuredListRow,
  StructuredListInput,
  StructuredListCell,
  StructuredListSkeleton,
} from './components/StructuredList';
export Tab from './components/Tab';
export TabContent from './components/TabContent';
export Tabs from './components/Tabs';
export Tag from './components/Tag';
export TextArea from './components/TextArea';
export TextInput from './components/TextInput';
export {
  Tile,
  ClickableTile,
  SelectableTile,
  ExpandableTile,
  TileAboveTheFoldContent,
  TileBelowTheFoldContent,
} from './components/Tile';
export RadioTile from './components/RadioTile';
export TileGroup from './components/TileGroup';
export TimePicker from './components/TimePicker';
export TimePickerSelect from './components/TimePickerSelect';
export Toggle from './components/Toggle';
export ToggleSmall from './components/ToggleSmall';
export Toolbar, {
  ToolbarItem,
  ToolbarTitle,
  ToolbarOption,
  ToolbarDivider,
} from './components/Toolbar';
export ToolbarSearch from './components/ToolbarSearch';
export Tooltip from './components/Tooltip';
export TooltipDefinition from './components/TooltipDefinition';
export TooltipIcon from './components/TooltipIcon';
export UnorderedList from './components/UnorderedList';
export SkeletonText from './components/SkeletonText';
export SkeletonPlaceholder from './components/SkeletonPlaceholder';
export DataTableSkeleton from './components/DataTableSkeleton';
export AccordionSkeleton from './components/Accordion/Accordion.Skeleton';
export BreadcrumbSkeleton from './components/Breadcrumb/Breadcrumb.Skeleton';
export ButtonSkeleton from './components/Button/Button.Skeleton';
export CheckboxSkeleton from './components/Checkbox/Checkbox.Skeleton';
export CodeSnippetSkeleton from './components/CodeSnippet/CodeSnippet.Skeleton';
export DropdownSkeleton from './components/Dropdown/Dropdown.Skeleton';
export FileUploaderSkeleton from './components/FileUploader/FileUploader.Skeleton';
export NumberInputSkeleton from './components/NumberInput/NumberInput.Skeleton';
export PaginationSkeleton from './components/Pagination/Pagination.Skeleton';
export ProgressIndicatorSkeleton from './components/ProgressIndicator/ProgressIndicator.Skeleton';
export RadioButtonSkeleton from './components/RadioButton/RadioButton.Skeleton';
export SearchSkeleton from './components/Search/Search.Skeleton';
export SelectSkeleton from './components/Select/Select.Skeleton';
export SliderSkeleton from './components/Slider/Slider.Skeleton';
export TabsSkeleton from './components/Tabs/Tabs.Skeleton';
export TagSkeleton from './components/Tag/Tag.Skeleton';
export TextAreaSkeleton from './components/TextArea/TextArea.Skeleton';
export TextInputSkeleton from './components/TextInput/TextInput.Skeleton';
export ToggleSkeleton from './components/Toggle/Toggle.Skeleton';
export ToggleSmallSkeleton from './components/ToggleSmall/ToggleSmall.Skeleton';
export IconSkeleton from './components/Icon/Icon.Skeleton';
export DatePickerSkeleton from './components/DatePicker/DatePicker.Skeleton';
export {
  Content,
  Header,
  HeaderContainer,
  HeaderGlobalAction,
  HeaderGlobalBar,
  HeaderMenu,
  HeaderMenuButton,
  HeaderMenuItem,
  HeaderName,
  HeaderNavigation,
  HeaderPanel,
  HeaderSideNavItems,
  Switcher,
  SwitcherItem,
  SwitcherDivider,
  SkipToContent,
  SideNav,
  SideNavDetails,
  SideNavDivider,
  SideNavFooter,
  SideNavHeader,
  SideNavIcon,
  SideNavItem,
  SideNavItems,
  SideNavLink,
  SideNavLinkText,
  SideNavMenu,
  SideNavMenuItem,
  SideNavSwitcher,
} from './components/UIShell';

// Experimental
export {
  FeatureFlags as unstable_FeatureFlags,
  useFeatureFlag as unstable_useFeatureFlag,
  useFeatureFlags as unstable_useFeatureFlags,
} from './components/FeatureFlags';
export {
  PageSelector as unstable_PageSelector,
  Pagination as unstable_Pagination,
} from './components/Pagination/experimental';
export unstable_TreeView, {
  TreeNode as unstable_TreeNode,
} from './components/TreeView';
export unstable_Menu, {
  MenuDivider as unstable_MenuDivider,
  MenuGroup as unstable_MenuGroup,
  MenuItem as unstable_MenuItem,
  MenuRadioGroup as unstable_MenuRadioGroup,
  MenuSelectableItem as unstable_MenuSelectableItem,
} from './components/Menu';
export { useContextMenu as unstable_useContextMenu } from './components/ContextMenu';
export {
  Heading as unstable_Heading,
  Section as unstable_Section,
} from './components/Heading';
export { default as unstable_ProgressBar } from './components/ProgressBar';
