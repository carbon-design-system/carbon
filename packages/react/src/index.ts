/**
 * Copyright IBM Corp. 2016, 2023
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import './feature-flags';

export * from './components/Accordion';
export * from './components/AccordionItem';
export * from './components/AspectRatio';
export * from './components/Breadcrumb';
export * from './components/Button';
export * from './components/ButtonSet';
export * from './components/Checkbox';
export * from './components/ClassPrefix';
export * from './components/CodeSnippet';
export * from './components/ComboBox';
export * from './components/ComposedModal';
export * from './components/ContainedList';
export * from './components/ContentSwitcher';
export * from './components/Copy';
export * from './components/CopyButton';
export * from './components/DangerButton';
export * from './components/DataTable';
export * from './components/DatePicker';
export * from './components/DatePickerInput';
export * from './components/Dropdown';
export * from './components/ErrorBoundary';
export * from './components/FileUploader';
export * from './components/FilterableMultiSelect';
export * from './components/Form';
export * from './components/FluidForm';
export * from './components/FormGroup';
export * from './components/FormItem';
export * from './components/FormLabel';
export * from './components/Grid';
export * from './components/IdPrefix';
export * from './components/InlineLoading';
export * from './components/Link';
export * from './components/ListItem';
export * from './components/Loading';
export * from './components/Modal';
export * from './components/ModalWrapper';
export * from './components/MultiSelect';
export * from './components/Notification';
export * from './components/NumberInput';
export * from './components/OrderedList';
export * from './components/OverflowMenu';
export * from './components/OverflowMenuItem';
export * from './components/Pagination';
export * from './components/PaginationNav';
export * from './components/PasswordInput';
export * from './components/PrimaryButton';
export * from './components/ProgressIndicator';
export * from './components/RadioButton';
export * from './components/RadioButtonGroup';
export * from './components/Search';
export * from './components/ExpandableSearch';
export * from './components/SecondaryButton';
export * from './components/Select';
export * from './components/SelectItem';
export * from './components/SelectItemGroup';
export * from './components/Switch';
export * from './components/Slider';
export * from './components/StructuredList';
export * from './components/Tab';
export * from './components/TabContent';
export * from './components/Tabs';
export * from './components/Tag';
export * from './components/TextArea';
export * from './components/TextInput';
export * from './components/Tile';
export * from './components/RadioTile';
export * from './components/TileGroup';
export * from './components/TimePicker';
export * from './components/TimePickerSelect';
export * from './components/Toggle';
export * from './components/Toggletip';
export * from './components/TreeView';
export * from './components/UnorderedList';
export * from './components/SkeletonText';
export * from './components/SkeletonPlaceholder';
export * from './components/SkeletonIcon';
export * from './components/DataTableSkeleton';
export * from './components/Pagination/Pagination.Skeleton';
export * from './components/RadioButton/RadioButton.Skeleton';
export * from './components/Tag/Tag.Skeleton';
export * from './components/Toggle/Toggle.Skeleton';
export * from './components/ToggleSmall/ToggleSmall.Skeleton';
export * from './components/Icon/Icon.Skeleton';
export * from './components/UIShell';

// Experimental
export { useContextMenu as unstable_useContextMenu } from './components/ContextMenu';
export {
  FeatureFlags as unstable_FeatureFlags,
  useFeatureFlag as unstable_useFeatureFlag,
  useFeatureFlags as unstable_useFeatureFlags,
} from './components/FeatureFlags';
export {
  FluidComboBox as unstable__FluidComboBox,
  FluidComboBoxSkeleton as unstable__FluidComboBoxSkeleton,
} from './components/FluidComboBox';
export {
  FluidDatePicker as unstable__FluidDatePicker,
  FluidDatePickerSkeleton as unstable__FluidDatePickerSkeleton,
} from './components/FluidDatePicker';
export { FluidDatePickerInput as unstable__FluidDatePickerInput } from './components/FluidDatePickerInput';
export {
  FluidDropdown as unstable__FluidDropdown,
  FluidDropdownSkeleton as unstable__FluidDropdownSkeleton,
} from './components/FluidDropdown';
export {
  FluidMultiSelect as unstable__FluidMultiSelect,
  FluidMultiSelectSkeleton as unstable__FluidMultiSelectSkeleton,
} from './components/FluidMultiSelect';
export {
  FluidSelect as unstable__FluidSelect,
  FluidSelectSkeleton as unstable__FluidSelectSkeleton,
} from './components/FluidSelect';
export {
  FluidTextArea as unstable__FluidTextArea,
  FluidTextAreaSkeleton as unstable__FluidTextAreaSkeleton,
} from './components/FluidTextArea';
export {
  FluidTextInput as unstable__FluidTextInput,
  FluidTextInputSkeleton as unstable__FluidTextInputSkeleton,
} from './components/FluidTextInput';
export {
  FluidTimePicker as unstable__FluidTimePicker,
  FluidTimePickerSkeleton as unstable__FluidTimePickerSkeleton,
} from './components/FluidTimePicker';
export { FluidTimePickerSelect as unstable__FluidTimePickerSelect } from './components/FluidTimePickerSelect';
export * from './components/Heading';
export * from './components/IconButton';
export * from './components/Layer';
export {
  LayoutDirection as unstable_LayoutDirection,
  useLayoutDirection as unstable_useLayoutDirection,
} from './components/Layout';
export {
  Menu as unstable_Menu,
  MenuItem as unstable_MenuItem,
  MenuItemDivider as unstable_MenuItemDivider,
  MenuItemGroup as unstable_MenuItemGroup,
  MenuItemRadioGroup as unstable_MenuItemRadioGroup,
  MenuItemSelectable as unstable_MenuItemSelectable,
} from './components/Menu';
export { OverflowMenuV2 as unstable_OverflowMenuV2 } from './components/OverflowMenuV2';
export {
  PageSelector as unstable_PageSelector,
  Pagination as unstable_Pagination,
} from './components/Pagination/experimental';
export * from './components/Popover';
export * from './components/ProgressBar';
export * from './components/Stack';
export * from './components/Tooltip';
export {
  Text as unstable_Text,
  TextDirection as unstable_TextDirection,
} from './components/Text';
export * from './components/Tooltip/DefinitionTooltip';
export * from './components/Theme';
export * from './internal/usePrefix';
export { useIdPrefix } from './internal/useIdPrefix';
