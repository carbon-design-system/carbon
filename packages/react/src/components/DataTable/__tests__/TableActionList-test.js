/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { render } from '@testing-library/react';
import React from 'react';
import { TableActionList } from '../';

describe('TableActionList', () => {
  it('should support a custom className on the outermost element', () => {
    const { container } = render(<TableActionList className="custom-class" />);
    expect(container.firstChild).toHaveClass('custom-class');
  });

  it('should spread props onto the outermost element', () => {
    const { container } = render(<TableActionList data-testid="test" />);
    expect(container.firstChild).toHaveAttribute('data-testid', 'test');
  });
});
