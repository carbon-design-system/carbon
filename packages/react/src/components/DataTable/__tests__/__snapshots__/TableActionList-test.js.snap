// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableActionList should render 1`] = `
<TableActionList
  className="custom-class"
>
  <div
    className="cds--action-list custom-class"
  />
</TableActionList>
`;
